doc/developing/os-compatibility.md

```diff
@@ -0,0 +1,94 @@
+OS/library compatibility policy
+===============================
+
+This document describes how we decide which versions of operating systems, C++ standards, libraries,
+build tools (CMake) or compilers Minetest requires as a minimum.
+
+As most development happens on Linux the first measure is to check the version of the component in question on:
+* the oldest still-supported **Ubuntu** (End of Standard Support)
+* the oldest still-supported **Debian** (*not* LTS)
+* optional: the second newest **RHEL (derivative)**
+
+Generally this leads to versions about 5 years old and works as a reasonable result for BSDs and other platforms too.
+
+Needless to say that any such changes need to work on our other platforms too, as listed below.
+
+### Windows
+
+We currently support Windows 8 or later.
+
+Despite requiring explicit support code in numerous places there doesn't seem to be a strong case for dropping older Windows versions. We will likely only do it once SDL2 does so.
+
+Note that we're constrained by the versions [vcpkg](https://vcpkg.io/en/packages) offers, for the MSVC build.
+
+### macOS
+
+We currently support macOS 10.14 (Mojave) or later.
+
+Since we do not have any macOS developer we can only do some shallow testing in CI.
+So this is subject to change basically whenever Github throws
+[a new version](https://github.com/actions/runner-images?tab=readme-ov-file#available-images) at us, or for other reasons.
+
+### Android
+
+We currently support Android 5.0 (API 21) or later.
+
+There's usually no reason to raise this unless the NDK drops older versions.
+Note: You can check the Google Play Console to see what our user base is running.
+
+## Other parts
+
+**Compilers**: gcc, clang and MSVC (exceptions exist)
+
+**OpenGL** is an entirely different beast, there is no current consensus on changing the requirements and neither do we have a exact set of requirements.
```

Comment:
>no current consensus on changing the requirements

I thought it was widely agreed upon that we go to opengl 3 (3.3?) as soon as needed.

(And "*an* exact".)

Reply:
sort of yeah...
will change to "formal consensus"

Reply:
After the sentence about OGL 1.4 being unreasonable, I would probably emphasize that we are considering dropping the fixed-function pipeline, in case it isn't clear.